client/blocks/payment-request/use-initialization.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState, useRef, useCallback } from '@wordpress/element';
+import { useStripe } from '@stripe/react-stripe-js';
+import { getSetting } from '@woocommerce/settings';
+import { __ } from '@wordpress/i18n';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoPaymentRequest,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptionSelectionsForCheckout,
+	getStripeServerData,
+	pluckAddress,
+	normalizeShippingOptions,
+} from '../stripe-utils';
+import { useEventHandlers } from './use-event-handlers';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ */
+
+export const useInitialization = ( {
+	billing,
+	shippingData,
+	setExpressPaymentError,
+	onClick,
+	onClose,
+	onSubmit,
+} ) => {
+	const stripe = useStripe();
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):void]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const [ isFinished, setIsFinished ] = useState( false );
+	const [ isProcessing, setIsProcessing ] = useState( false );
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const [ paymentRequestType, setPaymentRequestType ] = useState( '' );
+	const currentShipping = useRef( shippingData );
+	const {
+		paymentRequestEventHandlers,
+		clearPaymentRequestEventHandler,
+		setPaymentRequestEventHandler,
+	} = useEventHandlers();
+
+	// Update refs when any change.
+	useEffect( () => {
+		currentShipping.current = shippingData;
+	}, [ shippingData ] );
+
+	// Create the initial paymentRequest object. Note, we can't do anything if stripe isn't available yet or we have zero total.
+	useEffect( () => {
+		if (
+			! stripe ||
```

Comment:
Similar to a different comment earlier; I think it'd be in our interests to not rely on JS's type coercion, and instead explicitly check for `undefined !== stripe` or `null !== stripe`.

Reply:
I'm on the fence here. If we're going to explicitly check for type then it'd be better to check if `stripe instanceof` (or better yet use TS ;) )

Reply:
> or better yet use TS

That's _most definitely_ the ultimate goal 😛 

I'm on the fence about this a bit too, but decided to comment on it for discussion :)

Reply:
I think we can leave this as is for now. Marking this as resolved.